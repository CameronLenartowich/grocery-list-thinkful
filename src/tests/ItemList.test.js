import React from 'react';
import ReactDOM from 'react-dom';
import ItemListTest from '../components/ItemList';
import { exportAllDeclaration } from '@babel/types';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });
import * as firebase from 'firebase';
const firebaseConfig = {
    apiKey: "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
    authDomain: "grocery-list-1ed7b.firebaseapp.com",
    databaseURL: "https://grocery-list-1ed7b.firebaseio.com",
    projectId: "grocery-list-1ed7b",
    storageBucket: "grocery-list-1ed7b.appspot.com",
    messagingSenderId: "833209511385",
    appId: "1:833209511385:web:31f159d5098c2a96"
  };
  firebase.initializeApp(firebaseConfig);

const ItemList = () => 
    <ItemListTest
        firebase = {firebase}
        currentRoom= {1}
        currentRoomKey={1}
        user={"guest"}
    />;

// Snapshot

it('render correctly ItemList component', () => {  
    const component = renderer.create(<ItemList />).toJSON();
    expect(component).toMatchSnapshot();
});

// Props

/*
it("renders a todo item", () => {
    const component = Enzyme.mount(<ItemList fakeProp={"this is the fake prop"} />);
    console.log(component.props());
    expect((component).prop('fakeProp')).toEqual("guest");
})
*/

// trying contains() enzyme method

/*
    it("should contain static text", () => {
    const component = Enzyme.mount(<ItemList  
        firebase = {firebase}
        currentRoom= {1}
        currentRoomKey={1}
        user={"guest"}
        />);

        expect(component.contains(<h2>Select Conversation</h2>)).toEqual(true);
    })
*/

