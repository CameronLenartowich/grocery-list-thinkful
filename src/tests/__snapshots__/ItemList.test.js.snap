// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly ItemList component 1`] = `
<span>
  <h2
    className="todoTitle"
  >
    1
  </h2>
  <div
    className=""
  >
    <ul
      className="todos list-group"
    />
  </div>
  <form
    className="newTodo"
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <div
        className="col-10"
      >
        <input
          className="inputTodo form-control"
          id="newTodoInput"
          placeholder="text"
          type="text"
        />
      </div>
      <div
        className="col-2"
      >
        <button
          className="btn btn-block btn-outline-success"
          type="submit"
        >
          Add
        </button>
      </div>
    </div>
  </form>
</span>
`;
